openmpi & pmix: fix build & evaluation on Darwin

doronbehar opened the pull request:
## Description of changes


## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

doronbehar committed:
openmpi: fix configurePhase on darwin by always using our pmix

doronbehar committed:
openmpi: fix postInstall hook on Darwin

doronbehar commented:
@ofborg build pmix

nim65s commented:
Result of `nixpkgs-review pr 333016` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>220 packages marked as broken and skipped:</summary>
  <ul>
    <li>c3d</li>
    <li>ezminc</li>
    <li>frigate</li>
    <li>gplates</li>
    <li>gromacsCudaMpi</li>
    <li>gromacsCudaMpi.dev</li>
    <li>gromacsCudaMpi.man</li>
    <li>labplot</li>
    <li>librsb</li>
    <li>libtensorflow</li>
    <li>libtensorflow.python</li>
    <li>migrate</li>
    <li>nco</li>
    <li>netcdfcxx4</li>
    <li>python311Packages.bambi</li>
    <li>python311Packages.bambi.dist</li>
    <li>python311Packages.baselines</li>
    <li>python311Packages.baselines.dist</li>
    <li>python311Packages.cleanlab</li>
    <li>python311Packages.cleanlab.dist</li>
    <li>python311Packages.dalle-mini</li>
    <li>python311Packages.dalle-mini.dist</li>
    <li>python311Packages.distrax</li>
    <li>python311Packages.distrax.dist</li>
    <li>python311Packages.dm-haiku</li>
    <li>python311Packages.dm-haiku.dist</li>
    <li>python311Packages.dm-sonnet</li>
    <li>python311Packages.dm-sonnet.dist</li>
    <li>python311Packages.edward</li>
    <li>python311Packages.edward.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.evosax</li>
    <li>python311Packages.evosax.dist</li>
    <li>python311Packages.fastai</li>
    <li>python311Packages.fastai.dist</li>
    <li>python311Packages.fenics</li>
    <li>python311Packages.fenics.dist</li>
    <li>python311Packages.flax</li>
    <li>python311Packages.flax.dist</li>
    <li>python311Packages.flowmc</li>
    <li>python311Packages.flowmc.dist</li>
    <li>python311Packages.gpt-2-simple</li>
    <li>python311Packages.gpt-2-simple.dist</li>
    <li>python311Packages.imgcat</li>
    <li>python311Packages.imgcat.dist</li>
    <li>python311Packages.intensity-normalization</li>
    <li>python311Packages.intensity-normalization.dist</li>
    <li>python311Packages.islpy</li>
    <li>python311Packages.islpy.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
    <li>python311Packages.keras</li>
    <li>python311Packages.keras.dist</li>
    <li>python311Packages.llama-index</li>
    <li>python311Packages.llama-index-agent-openai</li>
    <li>python311Packages.llama-index-agent-openai.dist</li>
    <li>python311Packages.llama-index-cli</li>
    <li>python311Packages.llama-index-cli.dist</li>
    <li>python311Packages.llama-index-core</li>
    <li>python311Packages.llama-index-core.dist</li>
    <li>python311Packages.llama-index-embeddings-gemini</li>
    <li>python311Packages.llama-index-embeddings-gemini.dist</li>
    <li>python311Packages.llama-index-embeddings-google</li>
    <li>python311Packages.llama-index-embeddings-google.dist</li>
    <li>python311Packages.llama-index-embeddings-huggingface</li>
    <li>python311Packages.llama-index-embeddings-huggingface.dist</li>
    <li>python311Packages.llama-index-embeddings-ollama</li>
    <li>python311Packages.llama-index-embeddings-ollama.dist</li>
    <li>python311Packages.llama-index-embeddings-openai</li>
    <li>python311Packages.llama-index-embeddings-openai.dist</li>
    <li>python311Packages.llama-index-graph-stores-nebula</li>
    <li>python311Packages.llama-index-graph-stores-nebula.dist</li>
    <li>python311Packages.llama-index-graph-stores-neo4j</li>
    <li>python311Packages.llama-index-graph-stores-neo4j.dist</li>
    <li>python311Packages.llama-index-graph-stores-neptune</li>
    <li>python311Packages.llama-index-graph-stores-neptune.dist</li>
    <li>python311Packages.llama-index-indices-managed-llama-cloud</li>
    <li>python311Packages.llama-index-indices-managed-llama-cloud.dist</li>
    <li>python311Packages.llama-index-legacy</li>
    <li>python311Packages.llama-index-legacy.dist</li>
    <li>python311Packages.llama-index-llms-ollama</li>
    <li>python311Packages.llama-index-llms-ollama.dist</li>
    <li>python311Packages.llama-index-llms-openai</li>
    <li>python311Packages.llama-index-llms-openai-like</li>
    <li>python311Packages.llama-index-llms-openai-like.dist</li>
    <li>python311Packages.llama-index-llms-openai.dist</li>
    <li>python311Packages.llama-index-multi-modal-llms-openai</li>
    <li>python311Packages.llama-index-multi-modal-llms-openai.dist</li>
    <li>python311Packages.llama-index-program-openai</li>
    <li>python311Packages.llama-index-program-openai.dist</li>
    <li>python311Packages.llama-index-question-gen-openai</li>
    <li>python311Packages.llama-index-question-gen-openai.dist</li>
    <li>python311Packages.llama-index-readers-database</li>
    <li>python311Packages.llama-index-readers-database.dist</li>
    <li>python311Packages.llama-index-readers-file</li>
    <li>python311Packages.llama-index-readers-file.dist</li>
    <li>python311Packages.llama-index-readers-json</li>
    <li>python311Packages.llama-index-readers-json.dist</li>
    <li>python311Packages.llama-index-readers-llama-parse</li>
    <li>python311Packages.llama-index-readers-llama-parse.dist</li>
    <li>python311Packages.llama-index-readers-s3</li>
    <li>python311Packages.llama-index-readers-s3.dist</li>
    <li>python311Packages.llama-index-readers-twitter</li>
    <li>python311Packages.llama-index-readers-twitter.dist</li>
    <li>python311Packages.llama-index-readers-txtai</li>
    <li>python311Packages.llama-index-readers-txtai.dist</li>
    <li>python311Packages.llama-index-readers-weather</li>
    <li>python311Packages.llama-index-readers-weather.dist</li>
    <li>python311Packages.llama-index-vector-stores-chroma</li>
    <li>python311Packages.llama-index-vector-stores-chroma.dist</li>
    <li>python311Packages.llama-index-vector-stores-google</li>
    <li>python311Packages.llama-index-vector-stores-google.dist</li>
    <li>python311Packages.llama-index-vector-stores-postgres</li>
    <li>python311Packages.llama-index-vector-stores-postgres.dist</li>
    <li>python311Packages.llama-index-vector-stores-qdrant</li>
    <li>python311Packages.llama-index-vector-stores-qdrant.dist</li>
    <li>python311Packages.llama-index.dist</li>
    <li>python311Packages.llama-parse</li>
    <li>python311Packages.llama-parse.dist</li>
    <li>python311Packages.loopy</li>
    <li>python311Packages.loopy.dist</li>
    <li>python311Packages.medpy</li>
    <li>python311Packages.medpy.dist</li>
    <li>python311Packages.mhcflurry</li>
    <li>python311Packages.mhcflurry.dist</li>
    <li>python311Packages.n3fit</li>
    <li>python311Packages.n3fit.dist</li>
    <li>python311Packages.nanobind</li>
    <li>python311Packages.nanobind.dist</li>
    <li>python311Packages.objax</li>
    <li>python311Packages.objax.dist</li>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
    <li>python311Packages.pydicom-seg</li>
    <li>python311Packages.pydicom-seg.dist</li>
    <li>python311Packages.pygmt</li>
    <li>python311Packages.pygmt.dist</li>
    <li>python311Packages.pymc</li>
    <li>python311Packages.pymc.dist</li>
    <li>python311Packages.pymedio</li>
    <li>python311Packages.pymedio.dist</li>
    <li>python311Packages.pyopencl</li>
    <li>python311Packages.pyopencl.dist</li>
    <li>python311Packages.pyradiomics</li>
    <li>python311Packages.pyradiomics.dist</li>
    <li>python311Packages.pytensor</li>
    <li>python311Packages.pytensor.dist</li>
    <li>python311Packages.reikna</li>
    <li>python311Packages.reikna.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
    <li>python311Packages.sasmodels</li>
    <li>python311Packages.sasmodels.dist</li>
    <li>python311Packages.shimmy</li>
    <li>python311Packages.shimmy.dist</li>
    <li>python311Packages.skrl</li>
    <li>python311Packages.skrl.dist</li>
    <li>python311Packages.spacy</li>
    <li>python311Packages.spacy-loggers</li>
    <li>python311Packages.spacy-loggers.dist</li>
    <li>python311Packages.spacy-lookups-data</li>
    <li>python311Packages.spacy-lookups-data.dist</li>
    <li>python311Packages.spacy-transformers</li>
    <li>python311Packages.spacy-transformers.dist</li>
    <li>python311Packages.spacy.dist</li>
    <li>python311Packages.tensorflow</li>
    <li>python311Packages.tensorflow-build</li>
    <li>python311Packages.tensorflow-build.dist</li>
    <li>python311Packages.tensorflow-datasets</li>
    <li>python311Packages.tensorflow-datasets.dist</li>
    <li>python311Packages.tensorflow-probability</li>
    <li>python311Packages.tensorflow-probability.dist</li>
    <li>python311Packages.tensorflow.dist</li>
    <li>python311Packages.tensorflowWithoutCuda</li>
    <li>python311Packages.tensorflowWithoutCuda.dist</li>
    <li>python311Packages.textacy</li>
    <li>python311Packages.textacy.dist</li>
    <li>python311Packages.textnets</li>
    <li>python311Packages.textnets.dist</li>
    <li>python311Packages.tf-keras</li>
    <li>python311Packages.tf-keras.dist</li>
    <li>python311Packages.tf2onnx</li>
    <li>python311Packages.tf2onnx.dist</li>
    <li>python311Packages.tflearn</li>
    <li>python311Packages.tflearn.dist</li>
    <li>python311Packages.torchio</li>
    <li>python311Packages.torchio.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
    <li>python311Packages.trfl</li>
    <li>python311Packages.trfl.dist</li>
    <li>python311Packages.vqgan-jax</li>
    <li>python311Packages.vqgan-jax.dist</li>
    <li>python311Packages.wandb</li>
    <li>python311Packages.wandb.dist</li>
    <li>python311Packages.worldengine</li>
    <li>python311Packages.worldengine.dist</li>
    <li>python312Packages.fenics</li>
    <li>python312Packages.fenics.dist</li>
    <li>python312Packages.medpy</li>
    <li>python312Packages.medpy.dist</li>
    <li>python312Packages.pydicom-seg</li>
    <li>python312Packages.pydicom-seg.dist</li>
    <li>python312Packages.pygmt</li>
    <li>python312Packages.pygmt.dist</li>
    <li>python312Packages.pymedio</li>
    <li>python312Packages.pymedio.dist</li>
    <li>python312Packages.pyradiomics</li>
    <li>python312Packages.pyradiomics.dist</li>
    <li>python312Packages.shimmy</li>
    <li>python312Packages.shimmy.dist</li>
    <li>python312Packages.torchio</li>
    <li>python312Packages.torchio.dist</li>
    <li>python312Packages.worldengine</li>
    <li>python312Packages.worldengine.dist</li>
    <li>worldengine-cli</li>
    <li>worldengine-cli.dist</li>
    <li>xyce-parallel</li>
    <li>xyce-parallel.doc</li>
  </ul>
</details>
<details>
  <summary>233 packages failed to build:</summary>
  <ul>
    <li>ants</li>
    <li>bicgl</li>
    <li>bicpl</li>
    <li>cdo</li>
    <li>conglomerate</li>
    <li>dl-poly-classic-mpi</li>
    <li>ebtks</li>
    <li>eccodes</li>
    <li>elmerfem</li>
    <li>fftwMpi</li>
    <li>fftwMpi.dev</li>
    <li>fftwMpi.man</li>
    <li>gdal (python312Packages.gdal)</li>
    <li>grass</li>
    <li>gromacsDoubleMpi</li>
    <li>gromacsDoubleMpi.dev</li>
    <li>gromacsDoubleMpi.man</li>
    <li>gromacsMpi</li>
    <li>gromacsMpi.dev</li>
    <li>gromacsMpi.man</li>
    <li>haskellPackages.haskell-mpi</li>
    <li>haskellPackages.haskell-mpi.doc</li>
    <li>haskellPackages.hgdal</li>
    <li>haskellPackages.hgdal.doc</li>
    <li>hdf5-mpi</li>
    <li>hdf5-mpi.bin</li>
    <li>hdf5-mpi.dev</li>
    <li>highfive-mpi</li>
    <li>hp2p</li>
    <li>hpl</li>
    <li>hypre</li>
    <li>inormalize</li>
    <li>libminc</li>
    <li>libvdwxc</li>
    <li>mapcache</li>
    <li>mapnik</li>
    <li>mapproxy</li>
    <li>mapproxy.dist</li>
    <li>mapserver</li>
    <li>merkaartor</li>
    <li>minc_tools</li>
    <li>minc_widgets</li>
    <li>mni_autoreg</li>
    <li>mpi (openmpi)</li>
    <li>n3</li>
    <li>ncview</li>
    <li>nest-mpi</li>
    <li>netcdf</li>
    <li>netcdf-mpi</li>
    <li>netcdffortran</li>
    <li>neuron-full (python312Packages.neuronpy)</li>
    <li>neuron-mpi</li>
    <li>oobicpl</li>
    <li>openorienteering-mapper</li>
    <li>parmetis</li>
    <li>pdal</li>
    <li>perl536Packages.Tirex</li>
    <li>perl536Packages.Tirex.devdoc</li>
    <li>perl538Packages.Tirex</li>
    <li>perl538Packages.Tirex.devdoc</li>
    <li>precice</li>
    <li>pymol</li>
    <li>pymol.dist</li>
    <li>python311Packages.arviz</li>
    <li>python311Packages.arviz.dist</li>
    <li>python311Packages.boltztrap2</li>
    <li>python311Packages.boltztrap2.dist</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
    <li>python311Packages.cartopy</li>
    <li>python311Packages.cartopy.dist</li>
    <li>python311Packages.corner</li>
    <li>python311Packages.corner.dist</li>
    <li>python311Packages.dask-mpi</li>
    <li>python311Packages.dask-mpi.dist</li>
    <li>python311Packages.datashader</li>
    <li>python311Packages.datashader.dist</li>
    <li>python311Packages.django-bootstrap4</li>
    <li>python311Packages.django-bootstrap4.dist</li>
    <li>python311Packages.django-bootstrap5</li>
    <li>python311Packages.django-bootstrap5.dist</li>
    <li>python311Packages.eccodes</li>
    <li>python311Packages.fiona</li>
    <li>python311Packages.fiona.dist</li>
    <li>python311Packages.fipy</li>
    <li>python311Packages.fipy.dist</li>
    <li>python311Packages.folium</li>
    <li>python311Packages.folium.dist</li>
    <li>python311Packages.gdal</li>
    <li>python311Packages.geodatasets</li>
    <li>python311Packages.geodatasets.dist</li>
    <li>python311Packages.geopandas</li>
    <li>python311Packages.geopandas.dist</li>
    <li>python311Packages.gpaw</li>
    <li>python311Packages.gpaw.dist</li>
    <li>python311Packages.gstools</li>
    <li>python311Packages.gstools.dist</li>
    <li>python311Packages.h5netcdf</li>
    <li>python311Packages.h5netcdf.dist</li>
    <li>python311Packages.h5py-mpi</li>
    <li>python311Packages.h5py-mpi.dist</li>
    <li>python311Packages.inequality</li>
    <li>python311Packages.inequality.dist</li>
    <li>python311Packages.libpysal</li>
    <li>python311Packages.libpysal.dist</li>
    <li>python311Packages.mapclassify</li>
    <li>python311Packages.mapclassify.dist</li>
    <li>python311Packages.meshio</li>
    <li>python311Packages.meshio.dist</li>
    <li>python311Packages.momepy</li>
    <li>python311Packages.momepy.dist</li>
    <li>python311Packages.morecantile</li>
    <li>python311Packages.morecantile.dist</li>
    <li>python311Packages.mpi4py</li>
    <li>python311Packages.mpi4py.dist</li>
    <li>python311Packages.nest</li>
    <li>python311Packages.netcdf4</li>
    <li>python311Packages.netcdf4.dist</li>
    <li>python311Packages.neuronpy</li>
    <li>python311Packages.osmnx</li>
    <li>python311Packages.osmnx.dist</li>
    <li>python311Packages.plotnine</li>
    <li>python311Packages.plotnine.dist</li>
    <li>python311Packages.pyogrio</li>
    <li>python311Packages.pyogrio.dist</li>
    <li>python311Packages.pyprecice</li>
    <li>python311Packages.pyprecice.dist</li>
    <li>python311Packages.python-mapnik</li>
    <li>python311Packages.python-mapnik.dist</li>
    <li>python311Packages.qcodes</li>
    <li>python311Packages.qcodes-contrib-drivers</li>
    <li>python311Packages.qcodes-contrib-drivers.dist</li>
    <li>python311Packages.qcodes.dist</li>
    <li>python311Packages.rasterio</li>
    <li>python311Packages.rasterio.dist</li>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python311Packages.rioxarray</li>
    <li>python311Packages.rioxarray.dist</li>
    <li>python311Packages.sfepy</li>
    <li>python311Packages.sfepy.dist</li>
    <li>python311Packages.wktutils</li>
    <li>python311Packages.wktutils.dist</li>
    <li>python311Packages.wrf-python</li>
    <li>python311Packages.wrf-python.dist</li>
    <li>python312Packages.arviz</li>
    <li>python312Packages.arviz.dist</li>
    <li>python312Packages.boltztrap2</li>
    <li>python312Packages.boltztrap2.dist</li>
    <li>python312Packages.bsuite</li>
    <li>python312Packages.bsuite.dist</li>
    <li>python312Packages.cartopy</li>
    <li>python312Packages.cartopy.dist</li>
    <li>python312Packages.corner</li>
    <li>python312Packages.corner.dist</li>
    <li>python312Packages.dask-mpi</li>
    <li>python312Packages.dask-mpi.dist</li>
    <li>python312Packages.django-bootstrap4</li>
    <li>python312Packages.django-bootstrap4.dist</li>
    <li>python312Packages.django-bootstrap5</li>
    <li>python312Packages.django-bootstrap5.dist</li>
    <li>python312Packages.eccodes</li>
    <li>python312Packages.fiona</li>
    <li>python312Packages.fiona.dist</li>
    <li>python312Packages.folium</li>
    <li>python312Packages.folium.dist</li>
    <li>python312Packages.geodatasets</li>
    <li>python312Packages.geodatasets.dist</li>
    <li>python312Packages.geopandas</li>
    <li>python312Packages.geopandas.dist</li>
    <li>python312Packages.gpaw</li>
    <li>python312Packages.gpaw.dist</li>
    <li>python312Packages.gstools</li>
    <li>python312Packages.gstools.dist</li>
    <li>python312Packages.h5netcdf</li>
    <li>python312Packages.h5netcdf.dist</li>
    <li>python312Packages.h5py-mpi</li>
    <li>python312Packages.h5py-mpi.dist</li>
    <li>python312Packages.inequality</li>
    <li>python312Packages.inequality.dist</li>
    <li>python312Packages.libpysal</li>
    <li>python312Packages.libpysal.dist</li>
    <li>python312Packages.mapclassify</li>
    <li>python312Packages.mapclassify.dist</li>
    <li>python312Packages.meshio</li>
    <li>python312Packages.meshio.dist</li>
    <li>python312Packages.momepy</li>
    <li>python312Packages.momepy.dist</li>
    <li>python312Packages.morecantile</li>
    <li>python312Packages.morecantile.dist</li>
    <li>python312Packages.mpi4py</li>
    <li>python312Packages.mpi4py.dist</li>
    <li>python312Packages.nest</li>
    <li>python312Packages.netcdf4</li>
    <li>python312Packages.netcdf4.dist</li>
    <li>python312Packages.osmnx</li>
    <li>python312Packages.osmnx.dist</li>
    <li>python312Packages.plotnine</li>
    <li>python312Packages.plotnine.dist</li>
    <li>python312Packages.pyogrio</li>
    <li>python312Packages.pyogrio.dist</li>
    <li>python312Packages.pyprecice</li>
    <li>python312Packages.pyprecice.dist</li>
    <li>python312Packages.python-mapnik</li>
    <li>python312Packages.python-mapnik.dist</li>
    <li>python312Packages.qcodes</li>
    <li>python312Packages.qcodes-contrib-drivers</li>
    <li>python312Packages.qcodes-contrib-drivers.dist</li>
    <li>python312Packages.qcodes.dist</li>
    <li>python312Packages.rasterio</li>
    <li>python312Packages.rasterio.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
    <li>python312Packages.rioxarray</li>
    <li>python312Packages.rioxarray.dist</li>
    <li>python312Packages.sfepy</li>
    <li>python312Packages.sfepy.dist</li>
    <li>python312Packages.wktutils</li>
    <li>python312Packages.wktutils.dist</li>
    <li>python312Packages.wrf-python</li>
    <li>python312Packages.wrf-python.dist</li>
    <li>saga</li>
    <li>scalapack</li>
    <li>scalapack.dev</li>
    <li>t-rex</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi-c</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi-cxx</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi-f77</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi-f90</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.ompi-fort</li>
    <li>tests.pkg-config.defaultPkgConfigPackages.orte</li>
    <li>trilinos-mpi</li>
  </ul>
</details>

nim65s commented:
```
pmix> Running phase: fixupPhase
pmix> calling 'preFixup' function hook '_moveToShare'
pmix> calling 'preFixup' function hook '_multioutDocs'
pmix> calling 'preFixup' function hook '_multioutDevs'
pmix> Patching '/nix/store/wwra6sidx9pw12ckqzhsj857l21k6hnx-pmix-5.0.3-dev/lib/pkgconfig/pmix.pc' includedir to output /nix/store/wwra6sidx9pw12ckqzhsj857l21k6hnx-pmix-5.0.3-dev
pmix> evaling 'fixupOutput' string hook 'if [[ -z "${noAuditTmpdir-}" && -e "$prefix" ]]; then auditTmpdir "$prefix"; fi'
pmix> checking for references to /private/tmp/nix-build-pmix-5.0.3.drv-0/ in /nix/store/q7d2xxf60plkldkz26kgam1k6ri8gjzz-pmix-5.0.3...
pmix> evaling 'fixupOutput' string hook 'if [ -z "${dontGzipMan-}" ]; then compressManPages "$prefix"; fi'
pmix> calling 'fixupOutput' function hook '_moveLib64'
pmix> calling 'fixupOutput' function hook '_moveSbin'
pmix> calling 'fixupOutput' function hook '_moveSystemdUserUnits'
pmix> calling 'fixupOutput' function hook 'patchShebangsAuto'
pmix> patching script interpreter paths in /nix/store/q7d2xxf60plkldkz26kgam1k6ri8gjzz-pmix-5.0.3
pmix> calling 'fixupOutput' function hook '_pruneLibtoolFiles'
pmix> calling 'fixupOutput' function hook '_doStrip'
pmix> stripping (with command strip and flags -S) in  /nix/store/q7d2xxf60plkldkz26kgam1k6ri8gjzz-pmix-5.0.3/lib /nix/store/q7d2xxf60plkldkz26kgam1k6ri8gjzz-pmix-5.0.3/bin
pmix> evaling 'fixupOutput' string hook 'if [[ -z "${noAuditTmpdir-}" && -e "$prefix" ]]; then auditTmpdir "$prefix"; fi'
pmix> checking for references to /private/tmp/nix-build-pmix-5.0.3.drv-0/ in /nix/store/wwra6sidx9pw12ckqzhsj857l21k6hnx-pmix-5.0.3-dev...
pmix> evaling 'fixupOutput' string hook 'if [ -z "${dontGzipMan-}" ]; then compressManPages "$prefix"; fi'
pmix> calling 'fixupOutput' function hook '_moveLib64'
pmix> calling 'fixupOutput' function hook '_moveSbin'
pmix> calling 'fixupOutput' function hook '_moveSystemdUserUnits'
pmix> calling 'fixupOutput' function hook 'patchShebangsAuto'
pmix> patching script interpreter paths in /nix/store/wwra6sidx9pw12ckqzhsj857l21k6hnx-pmix-5.0.3-dev
pmix> calling 'fixupOutput' function hook '_pruneLibtoolFiles'
pmix> calling 'fixupOutput' function hook '_doStrip'
pmix> stripping (with command strip and flags -S) in  /nix/store/wwra6sidx9pw12ckqzhsj857l21k6hnx-pmix-5.0.3-dev/lib /nix/store/wwra6sidx9pw12ckqzhsj857l21k6hnx-pmix-5.0.3-dev/bin
pmix> evaling implicit 'postFixup' string hook
pmix> sed: no input files
pmix> /nix/store/28j5bkydq5fpiy2pllanssqdnk65an0z-stdenv-darwin/setup: line 203: pop_var_context: head of shell_variables not a function context

doronbehar committed:
pmix: use finalAttrs: pattern

doronbehar committed:
pmix: don't use with lib; in meta

doronbehar committed:
pmix: add doronbehar to maintainers

doronbehar committed:
pmix: nixfmt

doronbehar committed:
pmix: enable on all platforms

Disable multiple outputs on non Linux platforms, and don't perform
wrapper-data substitution on non-Linux platforms.

doronbehar committed:
pmix: use "''${!outputDev}" instead of $dev

This allows disabling multiple outputs more easily, as the first will
evaluate to $out if a "dev" output is not enabled.

doronbehar committed:
pmix: put substitution & wrapping in install & fixup respectively

Make it match the same pattern as in openmpi.

doronbehar commented:
> Result of `nixpkgs-review pr 333016` run on aarch64-darwin

Thanks for the review :) Hopefully these issues are fixed now.

doronbehar commented:
> Result of `nixpkgs-review pr 333016` run on aarch64-darwin

Oh and BTW be careful with nixpkgs-review on these packages - if openmpi won't fail, you probably won't have patience to build tensorflow packages... See also https://github.com/Mic92/nixpkgs-review/issues/281 .

doronbehar committed:
openmpi: 5.0.3 -> 5.0.5

nim65s commented:
I could see that 😅 
But in that case, shouldn't we target staging ?

 2a167e8411e95c0c7667140019598b63ab383427 also says:
```
openmpi> evaling implicit 'postInstall' string hook
openmpi> handling mpiCC
openmpi> handling mpic++
openmpi> handling mpicxx
openmpi> handling mpicc
openmpi> handling mpif77
openmpi> handling mpif90
openmpi> handling mpifort
openmpi> handling shmemCC
openmpi> handling shmemc++
openmpi> handling shmemcxx
openmpi> handling shmemcc
openmpi> handling shmemf77
openmpi> handling shmemf90
openmpi> handling shmemfort
openmpi> handling oshCC
openmpi> handling oshc++
openmpi> handling oshcxx
openmpi> handling oshcc
openmpi> handling oshf77
openmpi> handling oshf90
openmpi> handling oshfort
openmpi> substituteStream() in derivation openmpi-5.0.5: WARNING: pattern compiler=g++ doesn't match anything in file '/nix/store/qsc9z9ax0wjd0lhd6rmk7sgfx6vpi9is-openmpi-5.0.5/share/openmpi/mpic++-wrapper-data.txt'
openmpi> substituteStream() in derivation openmpi-5.0.5: WARNING: pattern compiler=gcc doesn't match anything in file '/nix/store/qsc9z9ax0wjd0lhd6rmk7sgfx6vpi9is-openmpi-5.0.5/share/openmpi/mpicc-wrapper-data.txt'
openmpi> substitute(): ERROR: file '/nix/store/qsc9z9ax0wjd0lhd6rmk7sgfx6vpi9is-openmpi-5.0.5/share/openmpi/shmemc++-wrapper-data.txt' does not exist
openmpi> /nix/store/28j5bkydq5fpiy2pllanssqdnk65an0z-stdenv-darwin/setup: line 203: pop_var_context: head of shell_variables not a function context

markuskowa commented:
> But in that case, shouldn't we target staging ?

For changes with a rebuild count below 500 this is not required (tensorflow is just a big package to build).

@doronbehar do you also want add yourself to the maintainer list of openmpi? 2 are more than 1.

doronbehar committed:
openmpi: add doronbehar to maintainers

doronbehar commented:
> @doronbehar do you also want add yourself to the maintainer list of openmpi? 2 are more than 1.

Sure :).

> [2a167e8](https://github.com/NixOS/nixpkgs/commit/2a167e8411e95c0c7667140019598b63ab383427) says:

Thanks again @nim65s for the report :) Now pushed another trial. I'll try to speed things up a bit if you are too busy to respond fast:

@ofborg build openmpi

doronbehar commented on pkgs/development/libraries/pmix/default.nix:
```diff
@@ -80,6 +81,7 @@ stdenv.mkDerivation (finalAttrs: {
     moveToOutput "bin/pmixcc" "''${!outputDev}"
     moveToOutput "share/pmix/pmixcc-wrapper-data.txt" "''${!outputDev}"
 
+  '' + lib.optionalString (lib.elem "dev" finalAttrs.outputs) ''
```
This is a mistake :) It should be conditioned on `stdenv.isLinux`, similarly to how it was done in `openmpi`... It won't change evaluation though.

doronbehar replied:
> This is a mistake :) It should be conditioned on `stdenv.isLinux`, similarly to how it was done in `openmpi`... It won't change evaluation though.

Done.

doronbehar committed:
pmix: nixfmt

markuskowa reviewed (approved):
Please wait with the merge until both darwin tests have passed (just to make sure we did not overlook anything).

doronbehar commented:
Hmm CI is so slow...

@ofborg build openmpi
